test/Microsoft.AspNetCore.Mvc.Core.Test/ModelBinding/BodyModelBinderTests.cs

```diff
@@ -73,7 +72,7 @@ public class BodyModelBinderTests
             Assert.False(binderResult.IsModelSet);
             Assert.Null(binderResult.Model);
 
-            // Key is empty because this was a top-level binding.
+            // Key is the fieldName because this was a top-level binding.
             var entry = Assert.Single(bindingContext.ModelState);
             Assert.Equal(string.Empty, entry.Key);
```

Comment:
Not really in the same way. You've got a definite model state error because the body failed to deserialize. 

The bug we're fixing is about missing validation errors. 

Let's talk more and see if we can solve both issues.